Kudu.Core/Functions/KedaFunctionTriggerProvider.cs

```diff
@@ -97,39 +100,53 @@ string ReplaceMatchedBindingExpression(Match match)
                 foreach (var metadata in scaleTrigger.Metadata)
                 {
                     var replacedValue = Regex.Replace(metadata.Value, Constants.AppSettingsRegex, matchEvaluator);
+                    
                     newMetadata[metadata.Key] = replacedValue;
                 }
 
                 scaleTrigger.Metadata = newMetadata;
             }
         }
 
-        public static IEnumerable<ScaleTrigger> GetFunctionTriggers(IEnumerable<JObject> functionsJson, string hostJsonText, IDictionary<string, string> appSettings)
+        public static IEnumerable<ScaleTrigger> GetFunctionTriggers(string appName, string appNamespace, IEnumerable<JObject> functionsJson, string hostJsonText)
         {
             var triggerBindings = functionsJson
                 .Select(o => ParseFunctionJson(o["functionName"]?.ToString(), o))
                 .SelectMany(i => i);
 
-            return CreateScaleTriggers(triggerBindings, hostJsonText, appSettings);
+            return CreateScaleTriggers(appName, appNamespace, triggerBindings, hostJsonText);
         }
 
-        public static IEnumerable<ScaleTrigger> GetFunctionTriggersFromSyncTriggerPayload(string synctriggerPayload,
-            IDictionary<string, string> appSettings)
+        public static IEnumerable<ScaleTrigger> GetFunctionTriggersFromSyncTriggerPayload(string appName, string appNamespace, string synctriggerPayload)
         {
-            return CreateScaleTriggers(ParseSyncTriggerPayload(synctriggerPayload), ParseHostJsonPayload(synctriggerPayload), appSettings);
+            return CreateScaleTriggers(appName, appNamespace, ParseSyncTriggerPayload(synctriggerPayload), ParseHostJsonPayload(synctriggerPayload));
         }
 
-        internal static IEnumerable<ScaleTrigger> CreateScaleTriggers(IEnumerable<FunctionTrigger> triggerBindings, string hostJsonText, IDictionary<string, string> appSettings)
+        internal static IEnumerable<ScaleTrigger> CreateScaleTriggers(string appName, string appNamespace, IEnumerable<FunctionTrigger> triggerBindings, string hostJsonText)
         {
-
             var durableTriggers = triggerBindings.Where(b => IsDurable(b));
             var standardTriggers = triggerBindings.Where(b => !IsDurable(b));
 
             var kedaScaleTriggers = new List<ScaleTrigger>();
             kedaScaleTriggers.AddRange(GetStandardScaleTriggers(standardTriggers));
 
-            // Update Binding Expression for %..% notation
-            UpdateFunctionTriggerBindingExpression(kedaScaleTriggers, appSettings);
+            try
+            {
+                var config = KubernetesClientConfiguration.InClusterConfig();
+                // Use the config object to create a client.
+                var client = new Kubernetes(config);
```

Comment:
Generally speaking, instantiate HTTP Client every time is not good idea. https://docs.microsoft.com/en-us/dotnet/api/system.net.http.httpclient?view=net-6.0 `Kubernetes`. We might pass the Http client for the object. 
In this case, we can't use `HttpClientFactory` so that having it as static might be a good idea.

Reply:
@SushmithaVReddy Can you address this feedback? I think it's important.